app/models/user.rb

```diff
@@ -258,8 +258,14 @@ def path
   end
 
   def followed_articles
-    Article
-      .cached_tagged_with_any(cached_followed_tag_names).unscope(:select)
+    relation = Article
+    if cached_antifollowed_tag_names.any?
+      relation = relation.not_cached_tagged_with_any(cached_antifollowed_tag_names)
+    end
+
+    relation
+      .cached_tagged_with_any(cached_followed_tag_names)
+      .unscope(:select)
       .union(Article.where(user_id: cached_following_users_ids))
```

Comment:
just pointing out (not a bug, but clarifying behavior as I read through this) - if a followed user adds an antifollowed tag to an article - it's shown in the digest, because of the union on followed users' articles.

Reply:
Yes, so this is a hard one. I figured the direct user follow overrides any tag preferences but I guess one could argue this point. That said, given how hard it is to figure out how to even antifollow a tag, I'm not sure how many of those we actually have. I tried to run some Blazer queries but they all timed out.